util/scfg/src/scfg.c

```diff
@@ -0,0 +1,254 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "os/mynewt.h"
+#include "scfg/scfg.h"
+
+#define SCFG_MAX_SETTING_ID_LEN    64
+#define SCFG_MAX_NUM_STR_LEN       (sizeof "18446744073709551615")
+
+#define SCFG_FOREACH_SETTING(group, s) \
+    for (s = (group)->settings; s->name != NULL; s++)
+
+static void
+scfg_setting_id(const char *group_name, const char *setting_name, char *buf)
+{
+    int setting_len;
+    int group_len;
+    int off;
+
+    group_len = strlen(group_name);
+    setting_len = strlen(setting_name);
+
+    /* <group>/<setting> */
+    assert(group_len + 1 + setting_len <= SCFG_MAX_SETTING_ID_LEN);
```

Comment:
Good catch, thanks.

I am making two changes:
1. Change that constant into a syscfg setting.
2. Allocate `max_len + 1` bytes to hold the setting ID (i.e., `max_len` refers to maximum string length, not buffer size).

Re: 2- I think that is better because it makes the system easier to configure.  The user can set the value to exactly the number of ascii characters required (and not have to think about null terminators).